cip/2.testable/CIP2015-06-24-call-procedures.adoc

```diff
@@ -177,21 +178,22 @@ If executing a procedure call causes any side effects (i.e. it "updates the grap
 before any results are returned to the user.
 An implementation may provide the user with a way to opt out of this behavior, however this must be done explicitly (e.g. via a configuration setting).
 
-==== Semantics of explicit argument passing
+==== Semantics of argument passing
 
-In this scenario, arguments are provided explicitly as a sequence of expressions as required by the procedure's signature.
+Arguments are provided as a sequence of expressions as required by the procedure's signature.
 It is an error if the number of provided arguments differs from the number of arguments required by the procedure signature.
 
 To call the procedure, all argument expressions are evaluated to argument values in order.
 It is an error if the argument values are incompatible with the argument types required by the procedure signature.
 
-==== Semantics of implicit argument passing
+=== About `YIELD *`
 
-In this scenario, arguments are provided implicitly via the parameters of the Cypher statement.
+`YIELD *` is only allowed in the standalone form of `CALL`.
+The reason why `YIELD *` is not allowed in the _In-query_ form of `CALL` is because it makes it hard for a reader of the query (and for query analysis tools) to reason about the scope of variables, since it does not make it explicit which variables are introduced, thus making later variable references in the query potentially ambiguous about whether they are introducing the variable or just referencing a variable introduced by `YIELD *`.
+To avoid these ambiguities `YIELD *` is disallowed within queries.
 
-To call the procedure, the argument values are obtained by using the parameter in scope with the same name as the procedure argument.
-If such a parameter does not exist, the argument value is taken to be `null`.
-It is an error, if the resulting argument values are incompatible with the argument types required by the procedure  signature.
+For queries that are made up solely by a single `CALL` though, `YIELD *` is not ambiguous, since there are no further clauses that could reference variables.
```

Comment:
I suggest rephrasing like so:

> However, for queries that are made up solely by a single `CALL`, `YIELD *` is not ambiguous since there are no further clauses that could reference variables.